kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/CustomResourceCrud1109Test.java

```diff
@@ -41,7 +41,7 @@
 
   @BeforeEach
   void setUp() {
-    customResourceDefinition = server.getClient().apiextensions().v1beta1().customResourceDefinitions()
+    /*customResourceDefinition = server.getClient().apiextensions().v1beta1().customResourceDefinitions()
```

Comment:
This test is explicit to verify that the arbitrary dashes in the names won't break the MockServer. It **shouldn't** be refactored.

Reply:
OK but it should still use the dashes in the CR name. I can revert it if needed.

Reply:
dashes are intended to be in the CRD (specifically in the plural names).
This is to verify that the crud mock server stores any created CR with a valid key.
It's better and clearer to keep the complete CRD definition within the test suite so that in a single view this is understood. However, I'll probably add an extra check to verify that the "dashed" path is preserved.

Reply:
Sorry, after re-checking the code, indeed it wasn't doing what it was supposed to… 😄 
I've changed it using `@Singular` and added more checks to verify that the generated CRD has indeed the desired properties.